modules/apps/29-fee/keeper/escrow.go

```diff
@@ -96,6 +96,10 @@ func (k Keeper) DistributePacketFeesOnTimeout(ctx sdk.Context, timeoutRelayer sd
 // If the distribution fails for any reason (such as the receiving address being blocked),
 // the state changes will be discarded.
 func (k Keeper) distributeFee(ctx sdk.Context, receiver sdk.AccAddress, fee sdk.Coins) {
+	if k.IsLocked(ctx) {
```

Comment:
I'm not sure how much extra computation this does. It is a little annoying we need to check if the fee module is locked for each single fee distribution, but otherwise we have to rework the API of all our other functions to properly abort a transaction without returning an error

Reply:
It's an extra state read. From a efficiency point of view, this will probs be cached so i wouldn't worry about it. But it is additional gas